torch/_functorch/_aot_autograd/jit_compile_runtime_wrappers.py

```diff
@@ -141,6 +141,27 @@ def rng_functionalization_wrapper(args):
     return compiled_fn
 
 
+# This checks to see if inductor generated different output strides compared to eager mode.
+# This mainly comes from inductor's layout optimzation, which can restride intermediates and outputs.
+# However, for dimensions that are size 1, it is ok to represent the corresponding stride with any value
+# (since that stride is not involved in the compute for any indexing into the tensor).
+# We don't want to report this case as "inductor diverges from eager",
+# since the divergence / layout optimization path does not support dynamic shapes
+# See https://github.com/pytorch/pytorch/issues/116433
+def strides_meaningfully_different(*, sizes, eager_strides, runtime_strides):
+    assert len(sizes) == len(eager_strides) == len(runtime_strides)
+    for size, eager_stride, runtime_stride in zip(
+        sizes, eager_strides, runtime_strides
+    ):
+        if size == 1:
+            if runtime_stride != eager_stride and runtime_stride != 0:
```

Comment:
If the size is 1, the stride can be really anything no? why the special case for 0 here?

Reply:
you're right (I'll change it). I just did zero since that's what inductor always picks ([here](https://github.com/pytorch/pytorch/blob/main/torch/_inductor/ir.py#L1975))

Reply:
Same for zero. The full condition can be summarised as
```python
    any(runtime_stride != eager_stride and size not in (0, 1) for size, eager_stride, runtime_stride in zip(sizes, eager_strides, runtime_strides)
```